service/worker/deployment/deployment_client.go

```diff
@@ -386,6 +387,7 @@ func (d *DeploymentClientImpl) generateRegisterWorkerInDeploymentArgs(
 		TaskQueueName:   taskQueueName,
 		TaskQueueType:   taskQueueType,
 		FirstPollerTime: timestamppb.New(firstPoll),
+		MaxTaskQueues:   int32(d.MaxTaskQueuesInDeployment(namespaceEntry.Name().String())),
```

Comment:
Ideally the limit should be passed to the workflow runner as a function that is can call in `validateRegisterWorker`. cc @dnr wdyt?

Reply:
otherwise we'd not be able to change the limit for deployments that are already created.

Reply:
oh nvm, this is added to the register args, not wf args. So it should be good. (still not having to pass it as update args would be preferable, because this is a config... but not a big deal)

Reply:
my thinking of passing it as an update arg was to propagate the change to the existing deployments asap, although I guess you can do with the runner too - imo, it's easier to read when passing the argument as an update arg since it's easier to follow but okay to change if majority thinks otherwise